save-plugins/fix-and-warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugins/fixandwarn/FixAndWarnPlugin.kt

```diff
@@ -0,0 +1,181 @@
+        filesAndTheirWarningsMap.forEach { (filePath, warningsList) ->
+            val fileData = fs.readLines(filePath) as MutableList
+            warningsList.forEach { (line, warningMsg) ->
+                fileData.add(line, warningMsg)
+            }
+            fs.write(filePath) {
+                fileData.forEach {
+                    write((it + "\n").encodeToByteArray())
+                }
+            }
+        }
+
+        // TODO: If we receive just one command for execution, then warn plugin should look at the fix plugin output
+        // TODO: for warnings, and not execute command one more time.
+        // TODO: Current approach works too, but in this case we have extra actions, which is not good.
+        // TODO: For the proper work it should be produced refactoring of warn plugin https://github.com/cqfn/save/issues/164,
+        // TODO: after which methods of warning comparison will be separated from the common logic
+        val warnTestResults = warnPlugin.handleFiles(testFilesAfterFix.asSequence())
+        return fixTestResults + warnTestResults
+    }
+
+    override fun rawDiscoverTestFiles(resourceDirectories: Sequence<Path>): Sequence<List<Path>> {
+        // Test files for fix and warn plugin should be the same, so this will be enough
+        return fixPlugin.rawDiscoverTestFiles(resourceDirectories)
+    }
+
+    /**
+     * Filter test resources
+     *
+     * @param suffix regex pattern of test resource
+     * @param match whether to keep elements which matches current pattern or to keep elements, which is not
+     * @return filtered list of files
+     */
+    private fun Sequence<List<Path>>.filterTestResources(suffix: Regex, match: Boolean): List<Path> {
+        val filteredFiles: MutableList<Path> = mutableListOf()
+        this.forEach { resources ->
+            filteredFiles.add(resources.single { path ->
+                if (match) {
+                    suffix.matchEntire(path.toString()) != null
+                } else {
+                    suffix.matchEntire(path.toString()) == null
+                }
+            })
+        }
+        return filteredFiles
+    }
+
+    /**
+     * Remove warnings from the given files, which satisfy pattern from [warn] plugin and save data about warnings, which were deleted
+     *
+     * @files files to be modified
+     *
+     * @return map of files and theirs list of warnings
+     */
+    private fun removeWarningsFromExpectedFiles(files: List<Path>): MutableMap<Path, WarningsList> {
+        val filesAndTheirWarningsMap: MutableMap<Path, WarningsList> = mutableMapOf()
+        files.forEach { file ->
+            val fileData = fs.readLines(file)
+            filesAndTheirWarningsMap[file] = mutableListOf()
+
+            val fileDataWithoutWarnings = fileData.filterIndexed { index, line ->
+                val isLineWithWarning = (warnPluginConfig.warningsInputPattern!!.find(line)?.groups != null)
+                if (isLineWithWarning) {
```

Comment:
You can chain `filterIndexed` and `onEach` to make it more clearly looking

Reply:
Seems it is not very suitable for current situation